core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -2186,6 +2186,13 @@ protected static Calendar getCalendarNotTooNear(int timeUnit) {
     checkConcat2Func(tester(SqlLibrary.ORACLE));
   }
 
+  @Test void testSubString() {
```

Comment:
It looks as if this function was never tested. Thanks for fixing the hole.

Rename the test method to `testOracleSubstrFunction` and move it next to `testSubstringFunction`.

Test a few more cases - e.g. indexes out of bounds, and null values, and CHAR argument.